### DISCUSSION TITLE
Climsoft Web First Release

### DISCUSSION THREAD

**Discussion by Patowhiz** [climsoft/climsoft-web#42] [Announcements]
@climsoft/climsoft-web-lead-developers 

As discussed in our annual catch-up meeting regarding the aims for **2024/5**, I suggest we target the completion of the following modules in the first release of the web application. 

<img width="397" alt="1" src="https://github.com/climsoft/climsoft-web/assets/21179007/039d7caa-80d4-4567-a2cc-b3a0c6af8f8c">

Moving forward, we should delve into specifics about what needs to be included in these modules for the initial release.

I am currently refining some implementation ideas for the quality control module, so expect some changes in the sub-modules shown in the diagram.

The first release will be designed to work in conjunction with the desktop database model via the export module. We anticipate making some modifications to the desktop application to facilitate seamless integration with the web application.

Furthermore, we need to discuss the web model and application in terms of its utility for regional services. Key features should include seamless data transfer between regional services and their member institutes. We also need to address the user management aspects of such integration.

**Comment by Patowhiz** [comment]
As discussed internally, the aim is a faster roll out of the web version without diminishing it's strengths. Without these strengths the significance of the web becomes extremely diminished. In summary,  the aims are in following order.

**First quarter:**
1. Discuss and test metadata modules(station, element, sources).
2. Discuss and test web form entry.
3. Discuss and test import entry.

**Second quarter:**
1. Develop, discuss and test Inventory module
2. Develop, discuss and test QC module.

**Third quarter:**
1. Develop, discuss and test Desktop data transfer.

This creates a room for continuous integration and testing because the entire modules share functionalities.

The above should be sufficient to have a prototype that we can demo to users and get feedback.

**Comment by smachua** [comment]
That's ok @Patowhiz. After our contracts are finalized we shall meet and agree on how to implement those activities.
